test/framework/kubernetes.go

```diff
@@ -280,3 +284,69 @@ func checkPodContainerHasEnvVariableWithValue(pod *corev1.Pod, containerName, en
 	}
 	return false
 }
+
+// GetIngressURI returns the ingress URI
+func GetIngressURI(namespace, serviceName string) (string, error) {
+	ingress := &v1beta1.Ingress{}
+	if exists, err := kubernetes.ResourceC(kubeClient).FetchWithKey(types.NamespacedName{Name: serviceName, Namespace: namespace}, ingress); err != nil {
+		return "", err
+	} else if !exists {
+		return "", fmt.Errorf("Ingress %s does not exist in namespace %s", serviceName, namespace)
+	} else if len(ingress.Spec.Rules) == 0 {
+		return "", fmt.Errorf("Ingress %s does not have any rules", serviceName)
+	}
+
+	return fmt.Sprintf("http://%s:80", ingress.Spec.Rules[0].Host), nil
+}
+
+// ExposeServiceOnKubernetes adds ingress CR to expose a service
+func ExposeServiceOnKubernetes(namespace, serviceName string) error {
+	host := serviceName
+	if !config.IsLocalCluster() {
+		host += fmt.Sprintf(".%s.%s", namespace, config.GetDomainSuffix())
```

Comment:
We should make sure that `GetDomainSuffix()` is defined in this place, to provide meaningful error in case user forgets to define the domain suffix.

Reply:
Good catch. I've added a verification stage before running the scenarios in order to check this kind of setup as I think it might be useful in other use cases. (See the new check_setup.go and my comments.